lib/puppet/provider/nameservice.rb

```diff
@@ -89,11 +89,19 @@ def section
       end
     end
 
-    def validate(name, value)
+    # This awful hack of passing `myself` is brought to you by the
+    # fact that this method isn't marked private, so I can't move it
+    # to be an instance method like I'd rather do - BER
+    def validate(name, value, myself=nil)
```

Comment:
This is the part I really hate.

The more I look at this the more I think we can toss it - it made way more sense when I was working on a different approach for this (where we were still going through the nameservice-generated setter). Now that I own the whole set/validation process in the useradd provider, I don't really need this anymore.